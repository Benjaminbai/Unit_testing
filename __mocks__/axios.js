
const mockAxios = jest.requireActual('Axios')

export default mockAxios;