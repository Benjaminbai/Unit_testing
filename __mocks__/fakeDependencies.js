jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useLocation: jest.fn().mockReturnValue({
            pathname: '',
            search: '',
            hash: '',
            state: null,
            key: '',
        }),
        useHistory: () => ({
            push: jest.fn(),
            location: {}
        }),
        Redirect: jest.fn().mockReturnValue({}),
    }
})
window.matchMedia = jest.fn().mockImplementation((query) => {
    const obj = {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
    };

    return obj;
})

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}))