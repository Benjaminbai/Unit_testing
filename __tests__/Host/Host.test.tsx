import '../../__mocks__/fakeDependencies'
import React from 'react'
import { mount, render } from 'enzyme'
import Host from '../../pages/Host/Host'
import { BrowserRouter as Router } from 'react-router-dom'

import { render as RTLRender, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import API from '../../api/Host'

describe('<Host />', () => {
    test('expect the Host exist', () => {
        const HostComponent = mount(<Router><Host /></Router>)
        expect(HostComponent.exists()).toBe(true)
    })

    test('expect Host unmount', () => {
        const HostComponent = mount(<Router><Host /></Router>)
        HostComponent.unmount()
        expect(() => {
            HostComponent.instance()
        }).toThrow()
    })

    // 快照
    test('the source code changed or not', () => {
        const HostComponent = render(<Router><Host /></Router>)
        expect(HostComponent).toMatchSnapshot()
    })

    test('the SearchForm component should be render', () => {
        const HostComponent = render(<Router><Host /></Router>)
        expect(HostComponent.find('form').length).toBe(1)
    })

    test('trigger detailRegistryHost api', () => {
        const spyFn = jest.spyOn(API, 'hostRegistryHost')
        RTLRender(<Router><Host /></Router>)
        userEvent.click(screen.getByText(/搜 索/i))
        expect(spyFn).toBeCalled()
    })

    test('detailRegistryHost api', () => {
        const spyFn = jest.spyOn(API, 'hostRegistryHost')
        RTLRender(<Router><Host /></Router>)
        expect(spyFn).toBeCalled()
    })
})