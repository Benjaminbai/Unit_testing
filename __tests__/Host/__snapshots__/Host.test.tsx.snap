// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Host /> the source code changed or not 1`] = `
<section
  class="ant-layout layout"
  style="height:100%"
>
  <header
    class="ant-layout-header header"
    style="background-color:#ffffff"
  >
    <div
      class="logo"
      style="margin:16px 0;width:200px"
    >
      <img
        alt="logo"
        width="120px"
      />
    </div>
    <div
      class=""
    >
      <ul
        class="ant-menu ant-menu-light right-container ant-menu-root ant-menu-horizontal"
        role="menu"
        style="float:right"
      >
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="display:none"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal"
          role="menuitem"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <div>
              <span
                aria-label="user"
                class="anticon anticon-user"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </span>
              -
            </div>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
          role="menuitem"
          style="visibility:hidden;position:absolute"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span>
              ···
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
    </div>
  </header>
  <section
    class="ant-layout"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark site-layout-background"
      style="flex:0 0 200px;max-width:200px;min-width:200px;width:200px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <ul
          class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
          role="menu"
          style="height:100%;border-right:0"
        >
          <li
            class="ant-menu-item"
            role="menuitem"
            style="padding-left:24px"
          >
            <span
              aria-label="home"
              class="anticon anticon-home"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
            <span>
              <a
                href="/dashboard"
              >
                Dashboard
              </a>
            </span>
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="aim"
                class="anticon anticon-aim"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="aim"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <defs />
                  <path
                    d="M952 474H829.8C812.5 327.6 696.4 211.5 550 194.2V72c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v122.2C327.6 211.5 211.5 327.6 194.2 474H72c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h122.2C211.5 696.4 327.6 812.5 474 829.8V952c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V829.8C696.4 812.5 812.5 696.4 829.8 550H952c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8zM512 756c-134.8 0-244-109.2-244-244s109.2-244 244-244 244 109.2 244 244-109.2 244-244 244z"
                  />
                  <path
                    d="M512 392c-32.1 0-62.1 12.4-84.8 35.2-22.7 22.7-35.2 52.7-35.2 84.8s12.5 62.1 35.2 84.8C449.9 619.4 480 632 512 632s62.1-12.5 84.8-35.2C619.4 574.1 632 544 632 512s-12.5-62.1-35.2-84.8A118.57 118.57 0 00512 392z"
                  />
                </svg>
              </span>
              <span>
                Agent
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="deployment-unit"
                class="anticon anticon-deployment-unit"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="deployment-unit"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M888.3 693.2c-42.5-24.6-94.3-18-129.2 12.8l-53-30.7V523.6c0-15.7-8.4-30.3-22-38.1l-136-78.3v-67.1c44.2-15 76-56.8 76-106.1 0-61.9-50.1-112-112-112s-112 50.1-112 112c0 49.3 31.8 91.1 76 106.1v67.1l-136 78.3c-13.6 7.8-22 22.4-22 38.1v151.6l-53 30.7c-34.9-30.8-86.8-37.4-129.2-12.8-53.5 31-71.7 99.4-41 152.9 30.8 53.5 98.9 71.9 152.2 41 42.5-24.6 62.7-73 53.6-118.8l48.7-28.3 140.6 81c6.8 3.9 14.4 5.9 22 5.9s15.2-2 22-5.9L674.5 740l48.7 28.3c-9.1 45.7 11.2 94.2 53.6 118.8 53.3 30.9 121.5 12.6 152.2-41 30.8-53.6 12.6-122-40.7-152.9zm-673 138.4a47.6 47.6 0 01-65.2-17.6c-13.2-22.9-5.4-52.3 17.5-65.5a47.6 47.6 0 0165.2 17.6c13.2 22.9 5.4 52.3-17.5 65.5zM522 463.8zM464 234a48.01 48.01 0 0196 0 48.01 48.01 0 01-96 0zm170 446.2l-122 70.3-122-70.3V539.8l122-70.3 122 70.3v140.4zm239.9 133.9c-13.2 22.9-42.4 30.8-65.2 17.6-22.8-13.2-30.7-42.6-17.5-65.5s42.4-30.8 65.2-17.6c22.9 13.2 30.7 42.5 17.5 65.5z"
                  />
                </svg>
              </span>
              <span>
                Master
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="tags"
                class="anticon anticon-tags"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="tags"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M483.2 790.3L861.4 412c1.7-1.7 2.5-4 2.3-6.3l-25.5-301.4c-.7-7.8-6.8-13.9-14.6-14.6L522.2 64.3c-2.3-.2-4.7.6-6.3 2.3L137.7 444.8a8.03 8.03 0 000 11.3l334.2 334.2c3.1 3.2 8.2 3.2 11.3 0zm62.6-651.7l224.6 19 19 224.6L477.5 694 233.9 450.5l311.9-311.9zm60.16 186.23a48 48 0 1067.88-67.89 48 48 0 10-67.88 67.89zM889.7 539.8l-39.6-39.5a8.03 8.03 0 00-11.3 0l-362 361.3-237.6-237a8.03 8.03 0 00-11.3 0l-39.6 39.5a8.03 8.03 0 000 11.3l243.2 242.8 39.6 39.5c3.1 3.1 8.2 3.1 11.3 0l407.3-406.6c3.1-3.1 3.1-8.2 0-11.3z"
                  />
                </svg>
              </span>
              <span>
                版本管理
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
          <li
            class="ant-menu-submenu ant-menu-submenu-inline"
            role="menuitem"
          >
            <div
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-menu-submenu-title"
              role="button"
              style="padding-left:24px"
            >
              <span
                aria-label="desktop"
                class="anticon anticon-desktop"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="desktop"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
                  />
                </svg>
              </span>
              <span>
                主机信息
              </span>
              <i
                class="ant-menu-submenu-arrow"
              />
            </div>
            <div />
          </li>
        </ul>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="padding:24px"
    >
      <div
        class="ant-breadcrumb"
        style="margin-bottom:16px"
      >
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            主机信息
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            主机列表
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
      </div>
      <main
        class="ant-layout-content site-layout-background"
        style="padding:20px;margin:0;min-height:280px"
      >
        <div
          style="height:100%;overflow-y:scroll;overflow-x:hidden"
        >
          <form
            class="ant-form ant-form-horizontal"
            style="padding:20px;background-color:#fbfbfb;border:1px solid #d9d9d9;margin-bottom:20px"
          >
            <div
              class="ant-row"
              style="margin-left:-12px;margin-right:-12px"
            >
              <div
                class="ant-col ant-col-6"
                style="padding-left:12px;padding-right:12px"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-5 ant-form-item-label"
                  >
                    <label
                      class=""
                      for="sys_name"
                      title="主机名"
                    >
                      主机名
                    </label>
                  </div>
                  <div
                    class="ant-col ant-col-19 ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="sys_name"
                          placeholder="请输入主机名"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-6"
                style="padding-left:12px;padding-right:12px"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-5 ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ip"
                      title="IP"
                    >
                      IP
                    </label>
                  </div>
                  <div
                    class="ant-col ant-col-19 ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="ip"
                          placeholder="请输入IP"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-6"
                style="padding-left:12px;padding-right:12px"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-5 ant-form-item-label"
                  >
                    <label
                      class=""
                      for="status"
                      title="状态"
                    >
                      状态
                    </label>
                  </div>
                  <div
                    class="ant-col ant-col-19 ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="status_list_0"
                                aria-autocomplete="list"
                                aria-controls="status_list"
                                aria-haspopup="listbox"
                                aria-owns="status_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="status"
                                readonly=""
                                role="combobox"
                                style="opacity:0"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-placeholder"
                            />
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select:none;-webkit-user-select:none"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-6"
                style="padding-left:12px;padding-right:12px"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-5 ant-form-item-label"
                  >
                    <label
                      class=""
                      for="machine_type"
                      title="主机类型"
                    >
                      主机类型
                    </label>
                  </div>
                  <div
                    class="ant-col ant-col-19 ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="machine_type_list_0"
                                aria-autocomplete="list"
                                aria-controls="machine_type_list"
                                aria-haspopup="listbox"
                                aria-owns="machine_type_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="machine_type"
                                readonly=""
                                role="combobox"
                                style="opacity:0"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-placeholder"
                            />
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select:none;-webkit-user-select:none"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row"
              style="margin-left:-12px;margin-right:-12px"
            >
              <div
                class="ant-col ant-col-6"
                style="padding-left:12px;padding-right:12px"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-5 ant-form-item-label"
                  >
                    <label
                      class=""
                      for="sys_kernel_ver"
                      title="内核版本"
                    >
                      内核版本
                    </label>
                  </div>
                  <div
                    class="ant-col ant-col-19 ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="sys_kernel_ver_list_0"
                                aria-autocomplete="list"
                                aria-controls="sys_kernel_ver_list"
                                aria-haspopup="listbox"
                                aria-owns="sys_kernel_ver_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="sys_kernel_ver"
                                readonly=""
                                role="combobox"
                                style="opacity:0"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-placeholder"
                            />
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select:none;-webkit-user-select:none"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-6"
                style="padding-left:12px;padding-right:12px"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-5 ant-form-item-label"
                  >
                    <label
                      class=""
                      for="sys_dist_ver"
                      title="系统版本"
                    >
                      系统版本
                    </label>
                  </div>
                  <div
                    class="ant-col ant-col-19 ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="sys_dist_ver_list_0"
                                aria-autocomplete="list"
                                aria-controls="sys_dist_ver_list"
                                aria-haspopup="listbox"
                                aria-owns="sys_dist_ver_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="sys_dist_ver"
                                readonly=""
                                role="combobox"
                                style="opacity:0"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-placeholder"
                            />
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select:none;-webkit-user-select:none"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-col ant-col-12"
                style="padding-left:12px;padding-right:12px;text-align:right"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="submit"
                >
                  <span>
                    搜 索
                  </span>
                </button>
              </div>
            </div>
          </form>
          <div
            class="ant-table-wrapper"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table ant-table-small ant-table-empty"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                    >
                      <table
                        style="table-layout:auto"
                      >
                        <colgroup />
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell"
                            >
                              主机名
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              IP
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              状态
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              系统
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              主机类型
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              CPU
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              内存(G)
                            </th>
                            <th
                              class="ant-table-cell"
                            >
                              磁盘(G)
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            class="ant-table-placeholder"
                          >
                            <td
                              class="ant-table-cell"
                              colspan="8"
                            >
                              <div
                                class="ant-empty ant-empty-normal"
                              >
                                <div
                                  class="ant-empty-image"
                                >
                                  <svg
                                    class="ant-empty-img-simple"
                                    height="41"
                                    viewBox="0 0 64 41"
                                    width="64"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <g
                                      fill="none"
                                      fill-rule="evenodd"
                                      transform="translate(0 1)"
                                    >
                                      <ellipse
                                        class="ant-empty-img-simple-ellipse"
                                        cx="32"
                                        cy="33"
                                        rx="32"
                                        ry="7"
                                      />
                                      <g
                                        class="ant-empty-img-simple-g"
                                        fill-rule="nonzero"
                                      >
                                        <path
                                          d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                        />
                                        <path
                                          class="ant-empty-img-simple-path"
                                          d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                        />
                                      </g>
                                    </g>
                                  </svg>
                                </div>
                                <p
                                  class="ant-empty-description"
                                >
                                  No Data
                                </p>
                              </div>
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</section>
`;
